docs/optimizations/FileLocalization.md

```diff
@@ -0,0 +1,52 @@
+# The 'localization_optional' Optimization
+
+## Scope
+
+The 'localization_optional' optimization applies to `File` and `File?` inputs on `task`s. It allows
+you to save time and money by perhaps not localizing some `File`s to the execution environment.
+
+## Effect
+
+The optimization signals to Cromwell that the task has been written in such a way that:
+ * The task will still work if the file were to be localized - say from a cloud bucket to the task's execution environment; but...
+ * The task will *also* work if the file is *not* localized prior to execution - and remains a path to a value in an object store, for example.  
```

Comment:
Suggested rephrase:
> AND
> * The task will *also* work if Cromwell does *not* localize the file 
>    * For example, with "localization_optional" enabled, a file will remain as a path to a cloud bucket. 

Questions:
* Why did you include the information "prior to execution"?
* Is a "cloud bucket" a more specific example of a "value in an object store"? If not, what's a clearer way of saying "path to a cloud bucket" or "value in object store"?

Reply:
* Artistic flourish. Will remove
* I'll change them to be consistently "in an object store"

Reply:
I think that this might be too string:
> For example, with "localization_optional" enabled, a file will remain as a path to a cloud bucket.

ie just because `localization_optional` is specified, it doesn't force Cromwell to not localize the file